vstart.sh: add PATH in vstart_environment

hualongfeng opened the pull request:
vstart.sh: add PATH in vstart_environment

Adding PATH environment variable can make the ceph command 
more convenient to use

Signed-off-by: Hualong Feng <hualong.feng@intel.com>

batrick commented:
jenkins test make check

hualongfeng commented:
jenkins test make check

tchaikov reviewed (changes requested):
> Adding PATH environment variable can make the ceph command more convenient to use

@hualongfeng could you be more specific ?  i have two concerns:

* adding `$PATH` here could pollute the environment in which we test ceph. for instance, we could end up using some other executables installed in `/usr/bin`. one might argue that `$CEPH_DIR/bin` is listed before existing `$PATH`, but what if some executable fails to build?
* a wrapper of ceph cli named `ceph_adm` is used for executing ceph commands. what's the advantage of using `ceph` directly?

hualongfeng commented:
> > Adding PATH environment variable can make the ceph command more convenient to use
> 
> @hualongfeng could you be more specific ? i have two concerns:
> 
> * adding `$PATH` here could pollute the environment in which we test ceph. for instance, we could end up using some other executables installed in `/usr/bin`. one might argue that `$CEPH_DIR/bin` is listed before existing `$PATH`, but what if some executable fails to build?
> * a wrapper of ceph cli named `ceph_adm` is used for executing ceph commands. what's the advantage of using `ceph` directly?

@tchaikov  I added this `$PATH` to start a cluster after vstart.sh, after source vstart_environemnt.sh script, we can directly use ceph-related commands, for example, directly use `rados -v` instead of `$CEPH_DIR/bin/rados -v`.

The `$PATH` added here only takes effect after `vstart.sh` starts a ceph cluster and manually source `vstart_environment.sh`, and only current shell to be effective. Whether to make the script effective depends on the user.

- Some executable fails to build. Doesn’t it mean that vstart.sh cannot start a ceph cluster normally, so it won’t generate the `vstart_environment.sh` script and won't source `vstart_environment.sh` script.

- The `ceph_adm` mentioned in the second concern seems to be the wrapper in `vstart.sh`. The `$PATH` I added is to facilitate the use of ceph-related commands in the shell not `vstart.sh` script after `vstart.sh` starts the cluster, and this `$PATH` is only effective in the current shell. This `$PATH` doesn't take effect in vstart.sh.

hualongfeng commented:
> > > Adding PATH environment variable can make the ceph command more convenient to use
> > 
> > 
> > @hualongfeng could you be more specific ? i have two concerns:
> > 
> > * adding `$PATH` here could pollute the environment in which we test ceph. for instance, we could end up using some other executables installed in `/usr/bin`. one might argue that `$CEPH_DIR/bin` is listed before existing `$PATH`, but what if some executable fails to build?
> > * a wrapper of ceph cli named `ceph_adm` is used for executing ceph commands. what's the advantage of using `ceph` directly?
> 
> @tchaikov I added this `$PATH` to start a cluster after vstart.sh, after source vstart_environemnt.sh script, we can directly use ceph-related commands, for example, directly use `rados -v` instead of `$CEPH_DIR/bin/rados -v`.
> 
> The `$PATH` added here only takes effect after `vstart.sh` starts a ceph cluster and manually source `vstart_environment.sh`, and only current shell to be effective. Whether to make the script effective depends on the user.
> 
> * Some executable fails to build. Doesn’t it mean that vstart.sh cannot start a ceph cluster normally, so it won’t generate the `vstart_environment.sh` script and won't source `vstart_environment.sh` script.
> * The `ceph_adm` mentioned in the second concern seems to be the wrapper in `vstart.sh`. The `$PATH` I added is to facilitate the use of ceph-related commands in the shell not `vstart.sh` script after `vstart.sh` starts the cluster, and this `$PATH` is only effective in the current shell. This `$PATH` doesn't take effect in vstart.sh.

And Develop mode need to set PATH, PYTHONPATH, LD_LIBRARY_PATH, however there are only PYTHONPATH, LD_LIBRARY_PATH to set in vstart_environemnt.sh

tchaikov commented:
> * Some executable fails to build. Doesn’t it mean that vstart.sh cannot start a ceph cluster normally, so it won’t generate the `vstart_environment.sh` script and won't source `vstart_environment.sh` script.

thanks for your detailed explanation. yeah, you are right. i thought that this `$PATH` would be consumed by `vstart.sh`. but as you explained, it's used by the vstart environment created by `vstart_environment.sh`, which is used by developer directly.

tchaikov commented:
jenkins test make check
